### ISSUE TITLE
Segmentation fault on `df.loc[k] = pd.Series({...})`

### ISSUE THREAD

**Issue by kuraga** [pandas-dev/pandas#17281]
#### Code Sample, a copy-pastable example if possible

```python
# Your code here
...
t = pd.Series({ ... })
df.loc[k] = t
```
#### Problem description

This code *sometimes* (at different loop iterations) causes a segmentation fault. Don't know which cases. Made [core dump](https://github.com/pandas-dev/pandas/files/1234725/core.python3.23708.1503066874.zip).

Test code (start several times): 
[test.zip](https://github.com/pandas-dev/pandas/files/2145091/test.zip)

#### Output of ``pd.show_versions()``

<details>
INSTALLED VERSIONS
------------------
commit: None
python: 3.6.5.final.0
python-bits: 64
OS: Linux
OS-release: 4.14.39-calculate
machine: x86_64
processor: Intel(R) Core(TM) i5-7200U CPU @ 2.50GHz
byteorder: little
LC_ALL: None
LANG: ru_RU.utf8
LOCALE: ru_RU.UTF-8

pandas: 0.23.0
pytest: None
pip: 9.0.3
setuptools: 39.0.1
Cython: None
numpy: 1.14.3
scipy: 1.1.0
pyarrow: None
xarray: None
IPython: 6.4.0
sphinx: None
patsy: 0.5.0
dateutil: 2.7.3
pytz: 2018.4
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: 2.2.2
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: 4.6.0
html5lib: 0.9999999
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
</details>

**Comment by TomAugspurger** [comment]
Can you provide a copy-pastable example?

**Comment by kuraga** [comment]
No (this code is a part of a big project). But I can try to reproduce...

**Comment by jorisvandenbossche** [comment]
>  But I can try to reproduce...

Please try (with some dummy dataframe that you can share). Otherwise there is not much we can do to help ..

**Comment by kuraga** [comment]
[test.zip](https://github.com/pandas-dev/pandas/files/1235263/test.zip) - but start *several times* :smile:

**Comment by TomAugspurger** [comment]
@kuraga can you update the original post with your reproducible example?

**Comment by kuraga** [comment]
@TomAugspurger done...

**Comment by TomAugspurger** [comment]
Thanks,

Does it rely on the specific contents of the file, or the fact that it's
being read from disk? Currently, we wouldn't be use that as a unit test,
since we don't want to include data files in the repo.

Ideally you would construct a dataframe by hand


```python
df = pd.DataFrame(...)
```

and then perform the operation that causes the segfault.

On Thu, Jun 28, 2018 at 6:38 AM, Alexander Kurakin <notifications@github.com

**Comment by kuraga** [comment]
@TomAugspurger ok, later

**Comment by kuraga** [comment]
@TomAugspurger closed in favour of #21968 .
